### PR TITLE
chore: bump find-my-way to 5.1.0

### PR DESCRIPTION
**Pull request by anonrig**
This pull request resolves https://github.com/fastify/fastify/issues/3399

#### Checklist

- [x] run `npm run test` and `npm run benchmark`
- [x] tests and/or benchmarks are included
- [ ] documentation is changed or added
- [x] commit message and code follows the [Developer's Certification of Origin](https://github.com/fastify/.github/blob/master/CONTRIBUTING.md#developers-certificate-of-origin-11)
      and the [Code of conduct](https://github.com/fastify/.github/blob/master/CODE_OF_CONDUCT.md)

### CODE REVIEW

File: test/hooks.test.js [hunk 1]
```diff
@@ -730,11 +739,11 @@ test('onRoute hook should able to change the route url', t => {
 
     sget({
       method: 'GET',
-      url: 'http://localhost:' + fastify.server.address().port + encodeURI('/föö')
+      url: getUrl(fastify) + encodeURI('/foo')
```
Comment (delvedor) [inline test/hooks.test.js:742]: Why this change?

**Review by delvedor — CHANGES_REQUESTED** [review]
Looks ok, a couple of notes.

File: test/hooks.test.js (hunk 1 above)
Comment (darkgl0w) [reply test/hooks.test.js:742]: @delvedor > It's related to the `dns` changes introduced with Node.js 17

**Review by mcollina — APPROVED** [review]
lgtm

File: test/hooks.test.js (hunk 1 above)
Comment (anonrig) [reply test/hooks.test.js:742]: the tests were failing with non-english character included router paths. if this is supported it should be handled in a different test since this is a test for changing the route.

**Comment by anonrig** [comment]
Are there any blocking issues for this pull request?

**Comment by jsumners** [comment]
> Are there any blocking issues for this pull request?

Yes. There are requested changes that have not been addressed.

File: test/hooks.test.js (hunk 1 above)
Comment (jsumners) [reply test/hooks.test.js:742]: Regardless, we have lost test coverage as a result of this change. It will be okay to move it to another test, but that test should be included as part of the change.

**Comment by anonrig** [comment]
@jsumners I've added the missing test. Thanks for the fast response!

**Review by jsumners — APPROVED** [review]
Looks good to me.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by darkgl0w — APPROVED** [review]

**Review by delvedor — APPROVED** [review]
LGTM
